.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,187 @@
+
+  gen_xml_doc:
+    name: Build XML documentation
+    runs-on: ubuntu-latest
+    container: greenbone/code-metrics-doxygen-debian-stretch
+    steps:
+      - name: Check out gvmd
+        uses: actions/checkout@v2
+      - name: Generate documentation (XML)
+        run: mkdir build && cd build/ && cmake -DSKIP_SRC=1 .. && make doc-xml 2> ~/doxygen-stderr.txt
+      - name: Upload doxygen error output as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: doxygen-stderr.txt
+          path: ~/doxygen-stderr.txt
+      - name: Upload XML documentation as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: xml-doc
+          path: build/doc/generated/xml/
+
+  build_gmp_doc:
+    name: Build GMP documentation
+    runs-on: ubuntu-latest
+    container: greenbone/build-env-gvm-master-debian-buster-clang-postgresql
+    steps:
+      - name: Check out gvmd
+        uses: actions/checkout@v2
+      - name: Generate GMP documentation (HTML)
+        run: mkdir build && cd build/ && cmake -DSKIP_SRC=1 .. && make doc-gmp
+      - name: Upload GMP documentation artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: gmp.html
+          path: build/doc/gmp.html
+
+  build:
+    name: Build various configs
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        build_type:
+          - Debug
+          - Release
+        docker_base:
+          - greenbone/build-env-gvm-master-debian-buster-clang-postgresql
+          - greenbone/build-env-gvm-master-debian-buster-gcc-postgresql
+    container: ${{ matrix.docker_base }}
+    env:
+      BUILD_TYPE: ${{ matrix.build_type }}
+    steps:
+      - name: Check out gvmd
+        uses: actions/checkout@v2
+      - name: Find gvmd reference branch
+        run: |
+          if [ -n "$GITHUB_BASE_REF" ];                          \
+          then echo "GVMD_REF=$GITHUB_BASE_REF" >> $GITHUB_ENV;  \
+          else echo "GVMD_REF=$GITHUB_REF" >> $GITHUB_ENV;       \
+          fi
+      - name: Select gvm-libs branch
+        run: |
+          if [ $GVMD_REF = "master" ];                           \
+            then GVM_LIBS_REF="master";                          \
+            else GVM_LIBS_REF=$(echo "$GVMD_REF"                 \
+                                | sed 's/refs\/.*\///g'          \
+                                | sed 's/gvmd-/gvm-libs-/g' -);  \
+            fi;                                                  \
+          echo "GVM_LIBS_REF=$GVM_LIBS_REF" >> $GITHUB_ENV
+      - name: Check out gvm-libs
+        uses: actions/checkout@v2
+        with:
+          repository: greenbone/gvm-libs
+          ref: ${{ env.GVM_LIBS_REF }}
+          path: gvm-libs
+      - name: Configure and compile gvm-libs (Debug)
+        run: bash -c "pushd gvm-libs && mkdir build && cd build/ && cmake -DCMAKE_BUILD_TYPE=Debug .. && make install && popd"
```

Comment:
You can insert `$BUILD_TYPE` here, too and use the same gvm-libs buildtype as gvmd?

Reply:
I've changed that in 4b4f5b6c5ec227aa3df8a067d91f2799bd201fd1